File: backend/GreenCart.Api/Models/ProductReview.cs
namespace GreenCart.Api.Models;

public sealed class ProductReview
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid OrderReviewId { get; set; }
    public OrderReview? OrderReview { get; set; }
    public Guid ProductId { get; set; }
    public Product? Product { get; set; }
    public int Rating { get; set; }
    public string? Comment { get; set; }
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
}

File: greencart_app/lib/src/features/catalog/models/product_review.dart
class ProductReview {
  const ProductReview({
    required this.id,
    required this.productId,
    required this.rating,
    this.comment,
    required this.createdAt,
  });

  final String id;
  final String productId;
  final int rating;
  final String? comment;
  final DateTime createdAt;

  factory ProductReview.fromJson(Map<String, dynamic> json) {
    return ProductReview(
      id: json['id'] as String,
      productId: json['productId'] as String,
      rating: json['rating'] as int,
      comment: json['comment'] as String?,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}
